### PR TITLE
Adding dropout for slot-specific values

### PR DESCRIPTION
**Pull request by emilysw**
Summary:
Sometimes, there may be overfitting on slot values. For example, in [IN:CREATE_CALL call [SL:CONTACT Mom]], the model often overfits on 'Mom' because 'Call Mom' is more common in the training data than calling other contacts.

This change applies a dropout layer to a token embedding (before concatenation with the dictionary features) if the last non-shift, non-reduction oracle (gold) action was a slot (and not an intent).

Differential Revision: D10843725

### CODE REVIEW

**Comment by Titousensei** [comment]
Abandoned
